packages/aila/src/core/llm/MockLLMService.ts

```diff
@@ -3,20 +3,43 @@ import { LLMService } from "./LLMService";
 export class MockLLMService implements LLMService {
   name = "MockLLM";
   private responseChunks: string[];
+  private responseObject: Record<string, unknown>;
 
-  constructor(responseChunks: string[] = ["This is ", "a mock ", "response."]) {
+  constructor(
+    responseChunks: string[] = ["This is ", "a mock ", "response."],
+    responseObject: Record<string, unknown> = {
+      response: "This is a response",
+      mock: true,
+      status: "complete",
+    },
+  ) {
     this.responseChunks = responseChunks;
+    this.responseObject = responseObject;
   }
 
   async createChatCompletionStream(): Promise<
-    ReadableStreamDefaultReader<Uint8Array | undefined>
+    ReadableStreamDefaultReader<string>
   > {
-    const encoder = new TextEncoder();
     const responseChunks = this.responseChunks;
     const stream = new ReadableStream({
       async start(controller) {
         for (const chunk of responseChunks) {
-          controller.enqueue(encoder.encode(chunk));
+          controller.enqueue(chunk);
+          await new Promise((resolve) => setTimeout(resolve, 4));
+        }
+        controller.close();
+      },
+    });
+    return stream.getReader();
+  }
+  async createChatCompletionObjectStream(): Promise<
```

Comment:
We now have two paths for completions: with and without objects. It's not clear to me from the code I've seen so far if we're in a transition period, if we need both, etc